### ISSUE TITLE
Doubt regarding src_key_padding_mask

### ISSUE THREAD

**Issue by thak123** [bentrevett/pytorch-seq2seq#76]
In the tranformer class
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/transformer.py
there is this field called src_key_padding_mask .

Can you please tell me why this field is used and what significance it has .

**Comment by bentrevett** [comment]
They use the `src_key_padding_mask` to mask the source sequence so the results of the Q and K matrix multiplications in the multi-headed attention layer will be masked out (i.e. zeroed) at some locations. You can use this, for example, to mask out any `<pad>` elements in the source sequence.

In the tutorials in this repo, we combine the key mask and the attention mask into one mask. In theory this is less computationally efficient as we are calculating a softmax over floats that we are going to just pad out anyway, and it would be better to pad them to zeros to speed up the softmax calculation. However, these repos are not focused on efficiency and are mainly to show the high level workings of certain algorithms/models.

**Comment by thak123** [comment]
@bentrevett Thank You
